src/test/ui-fulldeps/session-diagnostic/subdiagnostic-derive.rs

```diff
@@ -520,3 +518,122 @@ struct AZ {
     #[primary_span]
     span: Span,
 }
+
+#[derive(SessionSubdiagnostic)]
+#[suggestion(parser::add_paren, code = "...")]
+//~^ ERROR suggestion without `#[primary_span]` field
+struct BA {
+    #[suggestion_part]
+    //~^ ERROR `#[suggestion_part]` is not a valid attribute
```

Comment:
nit: in the rest of the file, the error annotations aren't indented

Reply:
Well, some of them aren't, a lot of them are - I decided to indent all the non-indented ones instead ;)